tasks/request_files/request_files.py

```diff
@@ -365,6 +347,40 @@ def inner_task(
         "asyncOperationId": event[EVENT_CONFIG_KEY][CONFIG_JOB_ID_KEY],
     }
 
+def get_glacier_recovery_type(config: Dict[str, Any]) -> str:
+    """
+    Returns the glacier recovery type from either config or environment variable. 
+    Must be either 'Bulk', 'Expedited', or 'Standard'.
+    Args:
+        config: The config dictionary from lambda event.
+
+    Raises: ValueError if recovery type value is invalid.
+    """
+
+
+    VALID_RESTORE_TYPES = ["Bulk", "Expedited", "Standard"]
+
+    # Look for config override
+    retrieval_type = config.get(CONFIG_RESTORE_DEFAULT_RECOVERY_TYPE_OVERRIDE_KEY, None)
+    if retrieval_type is not None:
+        if retrieval_type in VALID_RESTORE_TYPES:
+            LOGGER.info(f"Found glacier restore type from config: {retrieval_type}. Overridding other values.")
+        else:
+            LOGGER.error(f"Invalid value in restore type {retrieval_type} in config.")
```

Comment:
Possibly adjust error message from `f"Invalid value in restore type {retrieval_type} in config."` to `f"Invalid restore type value of {retrieval_type} found in the configuration {CONFIG_RESTORE_DEFAULT_RECOVERY_TYPE_OVERRIDE_KEY} key."` to allow the operator to better identify what configuration needs to be updated.

Reply:
updated